cf0b579412728d93d35301cd6e664dc5d9158e1f..210ddb614c64a0f0cca9fda17cd2938facb2237c

From 7592d0a40715b9c7b6949d888297a7cf59d0afbd Mon Sep 17 00:00:00 2001
From: Robert Walker <bob.mt.wya@gmail.com>
Date: Thu, 2 Mar 2023 11:45:57 +0000
Subject: [PATCH] configure.ac: fix 32-bit target build on 64-bit host

Fix Gentoo specific multilib Portage 32-bit build on 64-bit host
system.

See: https://bugs.gentoo.org/show_bug.cgi?id=395615

Explicitly add the required -m32/m64 to *FLAGS; this overrides any
arch-specific -m* flags that may have been appended by multilib-portage.

Even though -m32/m64 is now added to *FLAGS, -m32/m64 still has to be
explicitly added to CC and CXX due to wine's build system. For example,
winegcc saves the build-time value of CC and uses it at runtime.
---

 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)


diff --git a/configure.ac b/configure.ac
index 88e6b321d16..ae7a6587212 100644
--- a/configure.ac
+++ b/configure.ac
@@ -142,4 +142,7 @@ case $host in
       CC="$CC -m32"
       CXX="$CXX -m32"
+      CFLAGS="$CFLAGS -m32"
+      LDFLAGS="$LDFLAGS -m32"
+      CXXFLAGS="$CXXFLAGS -m32"
       AC_MSG_CHECKING([whether $CC works])
       AC_LINK_IFELSE([AC_LANG_PROGRAM()],AC_MSG_RESULT([yes]),
@@ -161,4 +164,7 @@ case $host in
       CC="$CC -m64"
       CXX="$CXX -m64"
+      CFLAGS="$CFLAGS -m64"
+      LDFLAGS="$LDFLAGS -m64"
+      CXXFLAGS="$CXXFLAGS -m64"
       host_cpu="x86_64"
       notice_platform="64-bit "
--
